src/main/resources/python/security_schemes.hbs

```diff
@@ -172,37 +231,175 @@ class MutualTLSSecurityScheme(__SecuritySchemeBase):
         body: typing.Optional[typing.Union[str, bytes]],
         query_params_suffix: typing.Optional[str],
         scope_names: typing.Tuple[str, ...] = (),
+{{#if oauthServerHostnames}}
+        oauth_server_client_info: OauthServerClientInfo = {}
+{{/if}}
     ) -> None:
         raise NotImplementedError("MutualTLSSecurityScheme not yet implemented")
 
 
+{{#if oauthServerHostnames}}
+class OAuthFlowBase:
+    def set_token(
+        self,
+        headers: _collections.HTTPHeaderDict,
+        token: OauthToken
+    ) -> None:
+        if token['token_type'].lower() == 'bearer':
+            headers.add('Authorization', f"Bearer {token['access_token']}")
+        else:
+            raise NotImplementedError(f"OauthToken token_type={token['token_type']} application not yet implemented")
+
+
 @dataclasses.dataclass
-class ImplicitOAuthFlow:
-    authorization_url: str
+class ImplicitOAuthFlow(OAuthFlowBase):
+    authorization_url: parse.ParseResult
     scopes: typing.Dict[str, str]
     refresh_url: typing.Optional[str] = None
+    
+    @property
+    def auth_or_token_url(self) -> parse.ParseResult:
+        return self.authorization_url
+
+    def apply_auth(
+        self,
+        headers: _collections.HTTPHeaderDict,
+        resource_path: str,
+        method: str,
+        body: typing.Optional[typing.Union[str, bytes]],
+        query_params_suffix: typing.Optional[str],
+        client_info: OauthClientInfo,
+        scope_names: typing.Tuple[str, ...] = ()
+    ) -> None:
+        """
+        Not implemented because this flow requires the user to visit a webpage and grant access
+        then get a redirection. The code would need to automatically do that.
+        """
+        raise NotImplementedError("ImplicitOAuthFlow not yet implemented")
+
+
+@dataclasses.dataclass
+class PasswordOauthFlow(OAuthFlowBase):
+    token_url: parse.ParseResult
+    scopes: typing.Dict[str, str]
+    refresh_url: typing.Optional[str] = None
+    username: str = ''
+    password: str = ''
```

Comment:
Why does this and password have a default, it must always be set by the developer when creating an instance. Please remove these defaults.

Reply:
Fixed